torch/distributed/_functional_collectives.py

```diff
@@ -291,6 +291,32 @@ def reduce_scatter_tensor_coalesced(
     return list(map(_maybe_wrap_tensor, tensor_list))
 
 
+def all_to_all_single(
```

Comment:
We should follow the design of other collectives and make the concat dim an explicit argument.

Reply:
@kumpera basically, adding these lines, is that right? https://github.com/pytorch/pytorch/blob/f4b464c6d0c090c4a29c4148ea289812f2720fea/torch/distributed/_functional_collectives.py#L175-L176

Reply:
The current design is actually sufficient for current use cases (and matches existing `dist.all_to_all_single` API). Maybe we can make adding concat dim a separate work item? (which would make a nice onboarding task too)

Reply:
We can add it later.
This is an old ask by @wanchaol as shuffling around collectives is a common source of bad overlap.